### PR TITLE
Email: Add functionality to email comparison page

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR updates the email comparison page by accounting for the possibility of two instead of three columns (i.e. when G Suite is not supported), as well as makes the buttons work.

#### Testing instructions

- Verify for a domain that doesn't support G Suite that we show only email forwarding and Titan (you can force this by passing `false` to `isGSuiteSupported`)
- Verify that the buttons work correctly for all providers (with Titan it should load a link from the backend)
- The label for the email forwarding button should say `Manage email forwarding` when the user has set up forwarding for that domain

### CODE REVIEW

File: client/my-sites/email/email-management/index.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/index.jsx:138]: Super minor but thought I could mention it: we usually name such variable `selectedDomain`.

File: client/my-sites/email/email-providers-comparison/email-provider-details/index.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-providers-comparison/email-provider-details/index.jsx:48]: Not mandatory at all but another way to handle missing event handlers could be to use a `noop` function:

```javascript
static defaultProps = {
  onButtonClick: noop,
};

onButtonClick = () => {
  this.props.onClick();
};
```

**Review by stephanethomas — COMMENTED** [review]
I had a quick look to those changes, and they look good to me. Let's wait for a proper review from @olaseni though 🙂.

File: client/my-sites/email/email-providers-comparison/email-provider-details/index.jsx
Comment (eltongo) [reply client/my-sites/email/email-providers-comparison/email-provider-details/index.jsx]: Yeah, this makes the code a lot cleaner.

**Review by olaseni — APPROVED** [review]
Looks good

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4489031

Hi @eltongo, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Manage email forwarding`
- `Annual billing`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4489031).

**Pull request merged**
